<h1>Table of Contents<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"></ul></div>

[Campaña DGT 2022](https://www.youtube.com/watch?v=690AqrQlVxs)

[web](https://los36.dgt.es/)

<img src="madrid_atasco.jpg" style="width:800px">

**Inicialización**

In [63]:
%matplotlib inline
import numpy as np
import pandas as pd
import plotly.express as px
import datetime as dt

Vamos a trabajar con el dataset del número de accidentes de tráfico en Madrid durante el 2019 (fuente: [https://datos.madrid.es/](https://datos.madrid.es/)). Fichero explicativo: `Estructura_DS_Accidentes_trafico_desde_2019.pdf`

In [64]:
# Comenzamos cargando el dataset de accidentes del 2019 en un dataframe que llamaremos df_2019.

df_2019 = pd.read_excel('2019_Accidentalidad.xlsx')
df_2019.head()

Nº  EXPEDIENTE       FECHA      HORA                              CALLE  \
0    2019S000020  01/01/2019  23:30:00                   CALL. FUENCARRAL   
1    2019S000017  01/01/2019  22:15:00           CALL. OCA / CALL. PINZON   
2    2019S000017  01/01/2019  22:15:00           CALL. OCA / CALL. PINZON   
3    2019S001812  01/01/2019  21:40:00  CALL. BAILEN / CUSTA. SAN VICENTE   
4    2019S001812  01/01/2019  21:40:00  CALL. BAILEN / CUSTA. SAN VICENTE   

  NÚMERO     DISTRITO           TIPO ACCIDENTE ESTADO METEREOLÓGICO  \
0    149     CHAMBERÍ                    Caída            Despejado   
1      -  CARABANCHEL  Colisión fronto-lateral            Despejado   
2      -  CARABANCHEL  Colisión fronto-lateral            Despejado   
3      -       CENTRO  Colisión fronto-lateral            Despejado   
4      -       CENTRO  Colisión fronto-lateral            Despejado   

  TIPO VEHÍCULO TIPO PERSONA       RANGO EDAD    SEXO  LESIVIDAD*  \
0    Ciclomotor    Conductor  DE 25 A 29 AÑOS  Hombre         1.0   
1       Turismo    Conductor  DE 40 A 44 AÑOS   Mujer        14.0   
2    Ciclomotor    Conductor  DE 35 A 39 AÑOS  Hombre         3.0   
3       Turismo    Conductor  DE 40 A 44 AÑOS  Hombre        14.0   
4       Turismo    Conductor  DE 30 A 34 AÑOS   Mujer         7.0   

   * La correspondencia de los códigos se encuentra descrito en la estructura del fichero.  
0                                                NaN                                        
1                                                NaN                                        
2                                                NaN                                        
3                                                NaN                                        
4                                                NaN

**Ejercicio 1:** ¿Quién está implicado en más accidentes de tráfico, los hombres o las mujeres? ¿Y en función del rango de edad? ¿Hay alguna franja de edad en la que las mujeres esten implicadas en más accidentes que los hombres?

In [65]:
# Comenzamos investigando que por el sexo de los implicados en accidentes.

df_2019.groupby('SEXO').count()['Nº  EXPEDIENTE']

SEXO
Hombre    31466
Mujer     15290
Name: Nº  EXPEDIENTE, dtype: int64

In [66]:
# Vamos a graficarlo ya que es más fácil de ver en este caso usaremos plotly.

px.bar(df_2019.groupby('SEXO').count()['Nº  EXPEDIENTE'].reset_index(), x='SEXO', y='Nº  EXPEDIENTE')

In [67]:
# Ahora vamos a ver el número de accidentes por rango de edad

df_2019.groupby('RANGO EDAD').count()['Nº  EXPEDIENTE']

RANGO EDAD
DE 0 A 5 AÑOS        520
DE 10 A 14 AÑOS      493
DE 15 A 17 AÑOS      344
DE 18 A 20 AÑOS     1534
DE 21 A 24 AÑOS     3350
DE 25 A 29 AÑOS     5610
DE 30 A 34 AÑOS     5463
DE 35 A 39 AÑOS     5573
DE 40 A 44 AÑOS     5736
DE 45 A 49 AÑOS     4955
DE 50 A 54 AÑOS     4258
DE 55 A 59 AÑOS     3265
DE 6 A 9 AÑOS        375
DE 60 A 64 AÑOS     2036
DE 65 A 69 AÑOS     1034
DE 70 A 74 AÑOS      731
DESCONOCIDA         5333
MAYOR DE 74 AÑOS    1196
Name: Nº  EXPEDIENTE, dtype: int64

In [68]:
# hacemos un plotly para verlo mejor

px.bar(df_2019.groupby('RANGO EDAD').count()['Nº  EXPEDIENTE'].reset_index(), x='RANGO EDAD', y='Nº  EXPEDIENTE', color='RANGO EDAD')

In [69]:
# En que rango de edad hay más accidentes de mujeres respecto a hombres

df_accidentes_mujeres = df_2019.query('SEXO == "Mujer"').groupby('RANGO EDAD').count()['Nº  EXPEDIENTE']
df_accidentes_hombres = df_2019.query('SEXO == "Hombre"').groupby('RANGO EDAD').count()['Nº  EXPEDIENTE']
diferencia_sexo_accidentes = df_accidentes_mujeres - df_accidentes_hombres
diferencia_sexo_accidentes

RANGO EDAD
DE 0 A 5 AÑOS        -46
DE 10 A 14 AÑOS        9
DE 15 A 17 AÑOS      -26
DE 18 A 20 AÑOS     -448
DE 21 A 24 AÑOS    -1166
DE 25 A 29 AÑOS    -1631
DE 30 A 34 AÑOS    -1812
DE 35 A 39 AÑOS    -2085
DE 40 A 44 AÑOS    -2308
DE 45 A 49 AÑOS    -1807
DE 50 A 54 AÑOS    -1770
DE 55 A 59 AÑOS    -1435
DE 6 A 9 AÑOS         -7
DE 60 A 64 AÑOS     -850
DE 65 A 69 AÑOS     -290
DE 70 A 74 AÑOS     -174
DESCONOCIDA         -112
MAYOR DE 74 AÑOS    -218
Name: Nº  EXPEDIENTE, dtype: int64

In [70]:
# hacemos un plotly por cada rango de edad y diferenciamos por sexo

px.bar(df_2019.groupby(['RANGO EDAD','SEXO']).count()['Nº  EXPEDIENTE'].reset_index(), x='RANGO EDAD', y='Nº  EXPEDIENTE', color='SEXO')

# Conclusiones ejercicio 1:

- Los hombres están implicados en más accidentes que las mujeres.
- Los hombres están implicados en más accidentes que las mujeres en todos los rangos de edad menos en el rango de edad de 10 a 14 años.
- En el rango de edad de 10 a 14 años hay más accidentes de mujeres que de hombres.

**Ejercicio 2:** Para el rango de edad para el cual hemos obtenido que hay más accidentes de tráfico con mujeres implicadas que con hombres, ¿esas mujeres eran conductoras, peatones o pasajeras? Para los casos en los que era conductora, ¿había alguna saltándose la ley y conduciendo un vehículo para el que necesitas carnet?

In [71]:
df_2019.head()

Nº  EXPEDIENTE       FECHA      HORA                              CALLE  \
0    2019S000020  01/01/2019  23:30:00                   CALL. FUENCARRAL   
1    2019S000017  01/01/2019  22:15:00           CALL. OCA / CALL. PINZON   
2    2019S000017  01/01/2019  22:15:00           CALL. OCA / CALL. PINZON   
3    2019S001812  01/01/2019  21:40:00  CALL. BAILEN / CUSTA. SAN VICENTE   
4    2019S001812  01/01/2019  21:40:00  CALL. BAILEN / CUSTA. SAN VICENTE   

  NÚMERO     DISTRITO           TIPO ACCIDENTE ESTADO METEREOLÓGICO  \
0    149     CHAMBERÍ                    Caída            Despejado   
1      -  CARABANCHEL  Colisión fronto-lateral            Despejado   
2      -  CARABANCHEL  Colisión fronto-lateral            Despejado   
3      -       CENTRO  Colisión fronto-lateral            Despejado   
4      -       CENTRO  Colisión fronto-lateral            Despejado   

  TIPO VEHÍCULO TIPO PERSONA       RANGO EDAD    SEXO  LESIVIDAD*  \
0    Ciclomotor    Conductor  DE 25 A 29 AÑOS  Hombre         1.0   
1       Turismo    Conductor  DE 40 A 44 AÑOS   Mujer        14.0   
2    Ciclomotor    Conductor  DE 35 A 39 AÑOS  Hombre         3.0   
3       Turismo    Conductor  DE 40 A 44 AÑOS  Hombre        14.0   
4       Turismo    Conductor  DE 30 A 34 AÑOS   Mujer         7.0   

   * La correspondencia de los códigos se encuentra descrito en la estructura del fichero.  
0                                                NaN                                        
1                                                NaN                                        
2                                                NaN                                        
3                                                NaN                                        
4                                                NaN

In [72]:
# Veamos para el rango de edad DE 10 A 14 AÑOS que tipo de persona estaba implicada en los accidentes

df_2019[(df_2019['SEXO']=='Mujer') & (df_2019['RANGO EDAD']=='DE 10 A 14 AÑOS')].groupby('TIPO PERSONA').count()['Nº  EXPEDIENTE']

TIPO PERSONA
Conductor      8
Pasajero     214
Peatón        29
Name: Nº  EXPEDIENTE, dtype: int64

In [73]:
# Vamos a crear una variable donde guardar el número de accidentes por tipo de persona

df_accidentes_mujeres_conductor = df_2019[(df_2019['SEXO']=='Mujer') & (df_2019['RANGO EDAD']=='DE 10 A 14 AÑOS') & (df_2019['TIPO PERSONA']=='Conductor')].groupby('TIPO VEHÍCULO').count()['Nº  EXPEDIENTE']
df_accidentes_mujeres_peaton = df_2019[(df_2019['SEXO']=='Mujer') & (df_2019['RANGO EDAD']=='DE 10 A 14 AÑOS') & (df_2019['TIPO PERSONA']=='Peatón')].groupby('TIPO VEHÍCULO').count()['Nº  EXPEDIENTE']
df_accidentes_mujeres_pasajero = df_2019[(df_2019['SEXO']=='Mujer') & (df_2019['RANGO EDAD']=='DE 10 A 14 AÑOS') & (df_2019['TIPO PERSONA']=='Pasajero')].groupby('TIPO VEHÍCULO').count()['Nº  EXPEDIENTE']

In [74]:
# y aqui tenemos los accidentes donde las mujeres eran conductores.

df_accidentes_mujeres_conductor

TIPO VEHÍCULO
Bicicleta                    6
Otros vehículos con motor    1
Sin especificar              1
Name: Nº  EXPEDIENTE, dtype: int64

In [75]:
# vamos a hacer un plotly del df_accidentes_mujeres_conductor para verlo mejor

px.bar(df_accidentes_mujeres_conductor.reset_index(), x='TIPO VEHÍCULO', y='Nº  EXPEDIENTE', color='TIPO VEHÍCULO')


# Conclusiones ejercicio 2:

- Los hombres están implicados en más accidentes que las mujeres.
- Los hombres están implicados en más accidentes que las mujeres en todos los rangos de edad menos en el rango de edad DE 10 A 14 AÑOS.
- En el rango de edad DE 10 A 14 AÑOS las mujeres están implicadas en más accidentes que los hombres.
- En el rango de edad DE 10 A 14 AÑOS las mujeres que estaban implicadas en accidentes eran: conductores, peatones y pasajeros.
- En el rango de edad DE 10 A 14 AÑOS las mujeres que estaban implicadas en accidentes como conductores eran un total de 8, 6 de ellas conducían bicicletas y 1 conducía algo a motor y por ultimo 1 de ellas sin especificar el tipo de vehículo. Sospechoso.... no?

**Ejercicio 3:** ¿Cuál fue el día del año en el que hubo más accidentes? Haz un plot del nº de accidentes por mes del año ¿Es el número de accidentes constante a lo largo de los meses de 2019?. _Hint: mismo Nº EXPEDIENTE implica que se trata del mismo accidente donde hay varios implicados y por lo tanto solo hay que contarlo una vez.

In [76]:
# Vamos a analizar en este caso los accidentes de 2019 para ver si hay algun dia del año en el que haya mas accidentes.

# primero vamos a eliminar los duplicados de Nº EXPEDIENTE

df_2019 = df_2019.drop_duplicates(subset='Nº  EXPEDIENTE')
df_2019.head()

Nº  EXPEDIENTE       FECHA      HORA  \
0     2019S000020  01/01/2019  23:30:00   
1     2019S000017  01/01/2019  22:15:00   
3     2019S001812  01/01/2019  21:40:00   
6     2019S001336  01/01/2019  20:25:00   
10    2019S000595  01/01/2019  20:03:00   

                                                CALLE NÚMERO     DISTRITO  \
0                                    CALL. FUENCARRAL    149     CHAMBERÍ   
1                            CALL. OCA / CALL. PINZON      -  CARABANCHEL   
3                   CALL. BAILEN / CUSTA. SAN VICENTE      -       CENTRO   
6                              CALL. MESONERO ROMANOS      1       CENTRO   
10  CALL. DOCTOR SANCHIS BANUS / CALL. DOCTOR CARM...      -        USERA   

                  TIPO ACCIDENTE ESTADO METEREOLÓGICO TIPO VEHÍCULO  \
0                          Caída            Despejado    Ciclomotor   
1        Colisión fronto-lateral            Despejado       Turismo   
3        Colisión fronto-lateral            Despejado       Turismo   
6            Atropello a persona            Despejado       Turismo   
10  Choque contra obstáculo fijo            Despejado       Turismo   

   TIPO PERSONA       RANGO EDAD    SEXO  LESIVIDAD*  \
0     Conductor  DE 25 A 29 AÑOS  Hombre         1.0   
1     Conductor  DE 40 A 44 AÑOS   Mujer        14.0   
3     Conductor  DE 40 A 44 AÑOS  Hombre        14.0   
6     Conductor  DE 18 A 20 AÑOS  Hombre        14.0   
10    Conductor      DESCONOCIDA     NaN         NaN   

    * La correspondencia de los códigos se encuentra descrito en la estructura del fichero.  
0                                                 NaN                                        
1                                                 NaN                                        
3                                                 NaN                                        
6                                                 NaN                                        
10                                                NaN

In [77]:
# Convertimos la columna FECHA a datetime

df_2019['FECHA'] = pd.to_datetime(df_2019['FECHA'])

/Users/adrianinfantesromero/opt/anaconda3/lib/python3.9/site-packages/pandas/core/tools/datetimes.py:1047: UserWarning:

Parsing '13/01/2019' in DD/MM/YYYY format. Provide format or specify infer_datetime_format=True for consistent parsing.

/Users/adrianinfantesromero/opt/anaconda3/lib/python3.9/site-packages/pandas/core/tools/datetimes.py:1047: UserWarning:

Parsing '14/01/2019' in DD/MM/YYYY format. Provide format or specify infer_datetime_format=True for consistent parsing.

/Users/adrianinfantesromero/opt/anaconda3/lib/python3.9/site-packages/pandas/core/tools/datetimes.py:1047: UserWarning:

Parsing '15/01/2019' in DD/MM/YYYY format. Provide format or specify infer_datetime_format=True for consistent parsing.

/Users/adrianinfantesromero/opt/anaconda3/lib/python3.9/site-packages/pandas/core/tools/datetimes.py:1047: UserWarning:

Parsing '16/01/2019' in DD/MM/YYYY format. Provide format or specify infer_datetime_format=True for consistent parsing.

/Users/adrianinfantesromero/opt/

In [78]:
df_2019["MES"] = df_2019["FECHA"].dt.month
df_2019["DIA"] = df_2019["FECHA"].dt.day
df_2019["DIA DEL AÑO"] = df_2019["FECHA"].dt.dayofyear

In [79]:
# comparamos los accidentes por cada mes el 2019

df_2019.groupby('MES').count()['Nº  EXPEDIENTE']

MES
1     1821
2     1675
3     1838
4     1755
5     2018
6     1845
7     1746
8     1465
9     1810
10    1954
11    2063
12    1945
Name: Nº  EXPEDIENTE, dtype: int64

In [80]:
# hacemos un plotly para verlo mejor

px.line(df_2019.groupby('MES').count()['Nº  EXPEDIENTE'].reset_index(), x='MES', y='Nº  EXPEDIENTE')

In [81]:
# evolucion de accidentes a lo largo del año 2019

px.line(df_2019.groupby('DIA DEL AÑO').count()['Nº  EXPEDIENTE'].reset_index(), x='DIA DEL AÑO', y='Nº  EXPEDIENTE')

In [82]:
# vamos a ver el día del año con más accidentes

df_2019.groupby('DIA DEL AÑO').count()['Nº  EXPEDIENTE'].sort_values(ascending=False).head(1)

DIA DEL AÑO
124    120
Name: Nº  EXPEDIENTE, dtype: int64

# Conclusiones ejercicio 3:

- El día del año con más accidentes fue el dia 124 del año 2019. Es decir el 23 de abril.
- El número de accidentes no es constante a lo largo de los meses del año 2019. Vemos que en los meses 5 y 11 son los picos de accidentes mientras que los meses 2 y 8 son los meses con menos accidentes.

**Ejercicio 4:** ¿Cuál fue el distrito con más accidentes? ¿y la calle? _Hint: de nuevo recordar que estamos hablando de nº de accidentes y el campo Nº EXPEDIENTE puede aparecer tantas veces como personas haya habido involucradas en el accidente.

In [83]:
# vamos a ver el distrito con más accidentes

df_2019['DISTRITO'].value_counts()

SALAMANCA              1736
PUENTE DE VALLECAS     1611
CHAMARTÍN              1608
CARABANCHEL            1346
CIUDAD LINEAL          1343
CENTRO                 1320
CHAMBERÍ               1209
MONCLOA-ARAVACA        1192
FUENCARRAL-EL PARDO    1111
SAN BLAS-CANILLEJAS    1107
RETIRO                 1086
LATINA                 1063
TETUÁN                 1048
ARGANZUELA              944
HORTALEZA               907
USERA                   742
MORATALAZ               626
VILLAVERDE              590
VILLA DE VALLECAS       556
BARAJAS                 481
VICÁLVARO               306
Name: DISTRITO, dtype: int64

In [84]:
px.bar(df_2019['DISTRITO'].value_counts().reset_index(), x='index', y='DISTRITO')

In [85]:
# vamos a ver las 7 calles con más accidentes

df_2019['CALLE'].value_counts().head(7)

PASEO. CASTELLANA                  309
CALL. ALCALA                       276
CALL. BRAVO MURILLO                122
AVDA. ALBUFERA                      95
CALL. SERRANO                       86
CALL. PRINCIPE DE VERGARA           80
PASEO. SANTA MARIA DE LA CABEZA     77
Name: CALLE, dtype: int64

In [86]:
px.bar(df_2019['CALLE'].value_counts().head(7).reset_index(), x='index', y='CALLE')

# Conclusiones ejercicio 4:

- El distrito con más accidentes es SALAMANCA.
- Las 7 calles con más accidentes son: PASEO. CASTELLANA, CALL. ALCALA, CALL. BRAVO MURILLO, AVDA. ALBUFERA, CALL. SERRANO, CALL. PRINCIPE DE VERGARA, PASEO. SANTA MARIA DE LA CABEZA.

**Ejercicio 5:** De las 7 calles con más accidentes del apartado anterior, genera una columna con la longitud de esa calle en metros o km y obten el nº de accidentes por longitud de la calle. Una vez hecho esto podemos ver cual es la calle más propensa a tener accidentes de las 7 seleccionadas.

In [87]:
longitud = pd.Series([6300,10500,4000,2500,1910,4200,3300],
                      index=['PASEO. CASTELLANA', 'CALL. ALCALA', 'CALL. BRAVO MURILLO', 'AVDA. ALBUFERA',
                      'CALL. SERRANO', 'CALL. PRINCIPE DE VERGARA', 'PASEO. SANTA MARIA DE LA CABEZA'] )

In [88]:
# añadimos la columna longitud a nuestro dataframe haciendolo coincidir con la calle indicada

df_2019['LONGITUD_MT'] = df_2019['CALLE'].map(longitud)

In [89]:
# Vamos a ver esas 7 calles con la longitud añadida

df_2019[df_2019['CALLE'].isin(longitud.index)]

Nº  EXPEDIENTE      FECHA      HORA                      CALLE NÚMERO  \
125      2019S000057 2019-02-01  19:05:00               CALL. ALCALA    233   
141      2019S000058 2019-02-01  17:14:00          PASEO. CASTELLANA    228   
144      2019S000068 2019-02-01  16:20:00          PASEO. CASTELLANA     66   
147      2019S001051 2019-02-01  16:00:00          PASEO. CASTELLANA    113   
167      2019S000941 2019-02-01  13:40:00          PASEO. CASTELLANA     81   
...              ...        ...       ...                        ...    ...   
51413    2019S039636 2019-12-28  11:50:00        CALL. BRAVO MURILLO    360   
51503    2019S039739 2019-12-29  12:45:00               CALL. ALCALA    676   
51694    2019S039824 2019-12-31  18:06:00        CALL. BRAVO MURILLO    302   
51774    2019S039843 2019-12-31  10:35:00               CALL. ALCALA    201   
51785    2019S039791 2019-12-31  07:35:00  CALL. PRINCIPE DE VERGARA      8   

                  DISTRITO           TIPO ACCIDENTE ESTADO METEREOLÓGICO  \
125              SALAMANCA         Colisión lateral                  NaN   
141              CHAMARTÍN      Atropello a persona            Despejado   
144              SALAMANCA                     Otro            Despejado   
147                 TETUÁN                  Alcance            Despejado   
167                 TETUÁN  Colisión fronto-lateral            Despejado   
...                    ...                      ...                  ...   
51413               TETUÁN         Colisión lateral            Despejado   
51503  SAN BLAS-CANILLEJAS         Colisión lateral            Despejado   
51694               TETUÁN                  Alcance            Despejado   
51774            SALAMANCA                  Alcance            Despejado   
51785            SALAMANCA  Colisión fronto-lateral            Despejado   

             TIPO VEHÍCULO TIPO PERSONA        RANGO EDAD    SEXO  LESIVIDAD*  \
125                Turismo    Conductor   DE 21 A 24 AÑOS  Hombre         NaN   
141                Turismo    Conductor  MAYOR DE 74 AÑOS  Hombre        14.0   
144                Turismo    Conductor   DE 55 A 59 AÑOS  Hombre         NaN   
147                Turismo    Conductor   DE 21 A 24 AÑOS   Mujer         NaN   
167    Motocicleta > 125cc    Conductor   DE 30 A 34 AÑOS  Hombre         7.0   
...                    ...          ...               ...     ...         ...   
51413              Turismo    Conductor   DE 40 A 44 AÑOS   Mujer         NaN   
51503              Turismo    Conductor   DE 25 A 29 AÑOS   Mujer         NaN   
51694              Turismo    Conductor   DE 45 A 49 AÑOS   Mujer         NaN   
51774              Turismo    Conductor   DE 35 A 39 AÑOS  Hombre         NaN   
51785              Turismo    Conductor   DE 30 A 34 AÑOS  Hombre        14.0   

       * La correspondencia de los códigos se encuentra descrito en la estructura del fichero.  \
125                                                  NaN                                         
141                                                  NaN                                         
144                                                  NaN                                         
147                                                  NaN                                         
167                                                  NaN                                         
...                                                  ...                                         
51413                                                NaN                                         
51503                                                NaN                                         
51694                                                NaN                                         
51774                                                NaN                                         
51785                                                NaN                                     

In [90]:
# creamos una tabla con el nombre de la calle, la longitud y el numero de accidentes

df_2019[df_2019['CALLE'].isin(longitud.index)].groupby('CALLE').agg({'LONGITUD_MT':'first', 'Nº  EXPEDIENTE':'count'})

LONGITUD_MT  Nº  EXPEDIENTE
CALLE                                                       
AVDA. ALBUFERA                        2500.0              95
CALL. ALCALA                         10500.0             276
CALL. BRAVO MURILLO                   4000.0             122
CALL. PRINCIPE DE VERGARA             4200.0              80
CALL. SERRANO                         1910.0              86
PASEO. CASTELLANA                     6300.0             309
PASEO. SANTA MARIA DE LA CABEZA       3300.0              77

In [91]:
# hacemos un plotly para verlo mejor

px.bar(df_2019[df_2019['CALLE'].isin(longitud.index)].groupby('CALLE').agg({'LONGITUD_MT':'first', 'Nº  EXPEDIENTE':'count'}).reset_index(), x='CALLE', y='Nº  EXPEDIENTE')


# Conclusiones ejercicio 5:

- La calle con más accidentes es PASEO. CASTELLANA con 6.300 metros de longitud y 309 accidentes.
- La calle con menos accidentes es PASEO. SANTA MARIA DE LA CABEZA con 3.300 metros de longitud y 77 accidente.

**Ejercicio 6:** Vamos a seleccionar los accidentes donde el tipo de vehículo implicado son las bicicletas. También disponemos del tráfico de bicicletas en Madrid durante el 2019 (BICICLETAS_2019.xlxs). Estudiar si existe algún tipo de correlación entre los distritos con más accidentes con bicicletas implicadas y los distritos donde más se usa la bicicleta (la comparativa puede ser a nivel nº total de accidentes con bicicleta en cada distrito en el año 2019 vs nº de bicicletas usadas por distrito en el año 2019).

In [92]:
# seleccionamos los accidentes donde el tipo de vehículo implicado son las bicicletas

accidentes_bicicletas_2019 = df_2019[df_2019['TIPO VEHÍCULO'] == 'Bicicleta']
accidentes_bicicletas_2019.head()

Nº  EXPEDIENTE      FECHA      HORA  \
29     2019S000659 2019-01-01  14:00:00   
274    2019S000132 2019-03-01  12:45:00   
282    2019S000045 2019-03-01  10:30:00   
683    2019S000904 2019-07-01  14:00:00   
755    2019S002373 2019-08-01  20:55:00   

                                                 CALLE NÚMERO  \
29         CALL. CASTELLO / CALL. DON RAMON DE LA CRUZ      -   
274        AVDA. SANTA EUGENIA / CALL. REAL DE ARGANDA      -   
282                           CTRA. DEHESA DE LA VILLA      1   
683  CALL. NUESTRA SEÑORA DE VALVERDE / CALL. DOCTO...      -   
755                      CALL. PUERTO DE LA CRUZ VERDE     49   

                DISTRITO           TIPO ACCIDENTE ESTADO METEREOLÓGICO  \
29             SALAMANCA                  Alcance            Despejado   
274    VILLA DE VALLECAS                  Alcance            Despejado   
282      MONCLOA-ARAVACA                    Caída            Despejado   
683  FUENCARRAL-EL PARDO  Colisión fronto-lateral            Despejado   
755           ARGANZUELA      Atropello a persona            Despejado   

    TIPO VEHÍCULO TIPO PERSONA       RANGO EDAD    SEXO  LESIVIDAD*  \
29      Bicicleta    Conductor  DE 25 A 29 AÑOS  Hombre         1.0   
274     Bicicleta    Conductor  DE 18 A 20 AÑOS  Hombre         7.0   
282     Bicicleta    Conductor  DE 15 A 17 AÑOS  Hombre         2.0   
683     Bicicleta    Conductor  DE 30 A 34 AÑOS  Hombre         7.0   
755     Bicicleta    Conductor  DE 45 A 49 AÑOS  Hombre        14.0   

     * La correspondencia de los códigos se encuentra descrito en la estructura del fichero.  \
29                                                 NaN                                         
274                                                NaN                                         
282                                                NaN                                         
683                                                NaN                                         
755                                                NaN                                         

     MES  DIA  DIA DEL AÑO  LONGITUD_MT  
29     1    1            1          NaN  
274    3    1           60          NaN  
282    3    1           60          NaN  
683    7    1          182          NaN  
755    8    1          213          NaN

In [93]:
# Vamos a modificar los valores de la columna DISTRITO para que coincidan con los del dataset de bicicletas con el fin de poder hacer un merge mediante un Title

accidentes_bicicletas_2019['DISTRITO'] = accidentes_bicicletas_2019['DISTRITO'].str.title()
accidentes_bicicletas_2019.head()

/var/folders/b7/wmcd_tzj3hd8lsdlfgr85gb40000gn/T/ipykernel_53810/674236430.py:3: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



Nº  EXPEDIENTE      FECHA      HORA  \
29     2019S000659 2019-01-01  14:00:00   
274    2019S000132 2019-03-01  12:45:00   
282    2019S000045 2019-03-01  10:30:00   
683    2019S000904 2019-07-01  14:00:00   
755    2019S002373 2019-08-01  20:55:00   

                                                 CALLE NÚMERO  \
29         CALL. CASTELLO / CALL. DON RAMON DE LA CRUZ      -   
274        AVDA. SANTA EUGENIA / CALL. REAL DE ARGANDA      -   
282                           CTRA. DEHESA DE LA VILLA      1   
683  CALL. NUESTRA SEÑORA DE VALVERDE / CALL. DOCTO...      -   
755                      CALL. PUERTO DE LA CRUZ VERDE     49   

                DISTRITO           TIPO ACCIDENTE ESTADO METEREOLÓGICO  \
29             Salamanca                  Alcance            Despejado   
274    Villa De Vallecas                  Alcance            Despejado   
282      Moncloa-Aravaca                    Caída            Despejado   
683  Fuencarral-El Pardo  Colisión fronto-lateral            Despejado   
755           Arganzuela      Atropello a persona            Despejado   

    TIPO VEHÍCULO TIPO PERSONA       RANGO EDAD    SEXO  LESIVIDAD*  \
29      Bicicleta    Conductor  DE 25 A 29 AÑOS  Hombre         1.0   
274     Bicicleta    Conductor  DE 18 A 20 AÑOS  Hombre         7.0   
282     Bicicleta    Conductor  DE 15 A 17 AÑOS  Hombre         2.0   
683     Bicicleta    Conductor  DE 30 A 34 AÑOS  Hombre         7.0   
755     Bicicleta    Conductor  DE 45 A 49 AÑOS  Hombre        14.0   

     * La correspondencia de los códigos se encuentra descrito en la estructura del fichero.  \
29                                                 NaN                                         
274                                                NaN                                         
282                                                NaN                                         
683                                                NaN                                         
755                                                NaN                                         

     MES  DIA  DIA DEL AÑO  LONGITUD_MT  
29     1    1            1          NaN  
274    3    1           60          NaN  
282    3    1           60          NaN  
683    7    1          182          NaN  
755    8    1          213          NaN

In [94]:
# Convertimos el dataset en un dataframe donde podemos observar el transito de bicis de cada distrito

transito__bicicletas_2019 = pd.read_excel('BICICLETAS_2019.xlsx')
transito__bicicletas_2019.head()

FECHA      HORA     IDENTIFICADOR  BICICLETAS  NÚMERO_DISTRITO  \
0 2019-01-01  00:00:00  PERM_BICI01_PM01           0                2   
1 2019-01-01  00:15:00  PERM_BICI01_PM01           0                2   
2 2019-01-01  00:30:00  PERM_BICI01_PM01           0                2   
3 2019-01-01  00:45:00  PERM_BICI01_PM01           0                2   
4 2019-01-01  01:00:00  PERM_BICI01_PM01           0                2   

      DISTRITO      NOMBRE_VIAL NÚMERO  CÓDIGO_POSTAL  \
0   Arganzuela  Calle de Toledo    178          28005   
1   Arganzuela  Calle de Toledo    178          28005   
2   Arganzuela  Calle de Toledo    178          28005   
3   Arganzuela  Calle de Toledo    178          28005   
4   Arganzuela  Calle de Toledo    178          28005   

                  OBSERVACIONES_DIRECCION    LATITUD  LONGITUD  
0  Sentido Gta. Pirámides Datos en prueba  40.401882 -3.713965  
1  Sentido Gta. Pirámides Datos en prueba  40.401882 -3.713965  
2  Sentido Gta. Pirámides Datos en prueba  40.401882 -3.713965  
3  Sentido Gta. Pirámides Datos en prueba  40.401882 -3.713965  
4  Sentido Gta. Pirámides Datos en prueba  40.401882 -3.713965

In [95]:
transito__bicicletas_2019['DISTRITO'] = transito__bicicletas_2019['DISTRITO'].str.title()
transito__bicicletas_2019.head()

FECHA      HORA     IDENTIFICADOR  BICICLETAS  NÚMERO_DISTRITO  \
0 2019-01-01  00:00:00  PERM_BICI01_PM01           0                2   
1 2019-01-01  00:15:00  PERM_BICI01_PM01           0                2   
2 2019-01-01  00:30:00  PERM_BICI01_PM01           0                2   
3 2019-01-01  00:45:00  PERM_BICI01_PM01           0                2   
4 2019-01-01  01:00:00  PERM_BICI01_PM01           0                2   

      DISTRITO      NOMBRE_VIAL NÚMERO  CÓDIGO_POSTAL  \
0   Arganzuela  Calle de Toledo    178          28005   
1   Arganzuela  Calle de Toledo    178          28005   
2   Arganzuela  Calle de Toledo    178          28005   
3   Arganzuela  Calle de Toledo    178          28005   
4   Arganzuela  Calle de Toledo    178          28005   

                  OBSERVACIONES_DIRECCION    LATITUD  LONGITUD  
0  Sentido Gta. Pirámides Datos en prueba  40.401882 -3.713965  
1  Sentido Gta. Pirámides Datos en prueba  40.401882 -3.713965  
2  Sentido Gta. Pirámides Datos en prueba  40.401882 -3.713965  
3  Sentido Gta. Pirámides Datos en prueba  40.401882 -3.713965  
4  Sentido Gta. Pirámides Datos en prueba  40.401882 -3.713965

In [96]:
# Vamos a hacer una pivot table para poder ver el numero de accidentes por cada distrito

pivot_accidentes_bicicletas_2019 = accidentes_bicicletas_2019.pivot_table(index='DISTRITO', values='Nº  EXPEDIENTE', aggfunc='count')
pivot_accidentes_bicicletas_2019

Nº  EXPEDIENTE
DISTRITO                           
Arganzuela                       46
Barajas                           9
Carabanchel                      22
Centro                           78
Chamartín                        13
Chamberí                         39
Ciudad Lineal                    15
Fuencarral-El Pardo              30
Hortaleza                        12
Latina                           13
Moncloa-Aravaca                  24
Moratalaz                         9
Puente De Vallecas               19
Retiro                           36
Salamanca                        39
San Blas-Canillejas              10
Tetuán                           23
Usera                            12
Vicálvaro                         4
Villa De Vallecas                 4
Villaverde                        9

In [97]:
# Hacemos lo mismo con el dataset de transito de bicis por distrito

pivot_transito_bicicletas_2019 = transito__bicicletas_2019.pivot_table(index='DISTRITO', values='IDENTIFICADOR', aggfunc='count')
pivot_transito_bicicletas_2019

IDENTIFICADOR
DISTRITO                           
 Arganzuela                   69275
 Carabanchel                  34852
 Centro                       38990
 Chamberí                     62922
 Ciudad Lineal                40250
 Fuencarral-El Pardo          63075
 Hortaleza                    55620
 Latina                       33440
 Moncloa-Aravaca              14343
 Moratalaz                    49549
 Puente De Vallecas           40588
 Salamanca                    95438

In [98]:
# usamos .index para poder hacer un merge

pivot_accidentes_bicicletas_2019.index

Index(['Arganzuela', 'Barajas', 'Carabanchel', 'Centro', 'Chamartín',
       'Chamberí', 'Ciudad Lineal', 'Fuencarral-El Pardo', 'Hortaleza',
       'Latina', 'Moncloa-Aravaca', 'Moratalaz', 'Puente De Vallecas',
       'Retiro', 'Salamanca', 'San Blas-Canillejas', 'Tetuán', 'Usera',
       'Vicálvaro', 'Villa De Vallecas', 'Villaverde'],
      dtype='object', name='DISTRITO')

In [99]:
pivot_transito_bicicletas_2019.index

Index([' Arganzuela', ' Carabanchel', ' Centro', ' Chamberí', ' Ciudad Lineal',
       ' Fuencarral-El Pardo', ' Hortaleza', ' Latina', ' Moncloa-Aravaca',
       ' Moratalaz', ' Puente De Vallecas', ' Salamanca'],
      dtype='object', name='DISTRITO')

In [100]:
# Podemos observar que el index de pivot_transito_bicicletas_2019 no coincide con el de pivot_accidentes_bicicletas_2019. Vamos a modificarlo para que coincida quitando los espacios en blanco

pivot_transito_bicicletas_2019.index = pivot_transito_bicicletas_2019.index.str.strip() # quitamos los espacios en blanco
pivot_accidentes_bicicletas_2019.index = pivot_accidentes_bicicletas_2019.index.str.strip() # quitamos los espacios en blanco

In [101]:
# Vamos a hacer un merge para poder comparar los datos

pivot_correlacion_transito_accidentes_bicicletas_2019 = pd.merge(pivot_accidentes_bicicletas_2019, pivot_transito_bicicletas_2019, on='DISTRITO')
pivot_correlacion_transito_accidentes_bicicletas_2019.columns = ['ACCIDENTES', 'TRANSITO']
pivot_correlacion_transito_accidentes_bicicletas_2019.head()

ACCIDENTES  TRANSITO
DISTRITO                           
Arganzuela             46     69275
Carabanchel            22     34852
Centro                 78     38990
Chamberí               39     62922
Ciudad Lineal          15     40250

In [102]:
# A este dataframe le vamos a añadir la columna de porcentaje de accidentes con bicicletas implicadas

pivot_correlacion_transito_accidentes_bicicletas_2019['PORCENTAJE_ACCIDENTES'] = pivot_correlacion_transito_accidentes_bicicletas_2019['ACCIDENTES'] / pivot_correlacion_transito_accidentes_bicicletas_2019['ACCIDENTES'].sum() * 100
pivot_correlacion_transito_accidentes_bicicletas_2019

ACCIDENTES  TRANSITO  PORCENTAJE_ACCIDENTES
DISTRITO                                                        
Arganzuela                   46     69275              13.294798
Carabanchel                  22     34852               6.358382
Centro                       78     38990              22.543353
Chamberí                     39     62922              11.271676
Ciudad Lineal                15     40250               4.335260
Fuencarral-El Pardo          30     63075               8.670520
Hortaleza                    12     55620               3.468208
Latina                       13     33440               3.757225
Moncloa-Aravaca              24     14343               6.936416
Moratalaz                     9     49549               2.601156
Puente De Vallecas           19     40588               5.491329
Salamanca                    39     95438              11.271676

In [118]:
# Hagamos un plotly de la pivot table

px.bar(pivot_correlacion_transito_accidentes_bicicletas_2019.reset_index(), x='DISTRITO', y='PORCENTAJE_ACCIDENTES')

In [112]:
# primero vamos a quitar los espacios en blanco de la columna DISTRITO del dataframe transito__bicicletas_2019

transito__bicicletas_2019['DISTRITO'] = transito__bicicletas_2019['DISTRITO'].str.strip()

In [114]:
# Creamos un diccionario con los distritos y la latitud y longitud de cada uno

diccionario_distritos = {}

for i in range(len(transito__bicicletas_2019)):
    diccionario_distritos[transito__bicicletas_2019['DISTRITO'][i]] = [transito__bicicletas_2019['LATITUD'][i], transito__bicicletas_2019['LONGITUD'][i]]

diccionario_distritos

{'Arganzuela': [40.3930232566823, -3.69756666109268],
 'Chamberí': [40.4280863873349, -3.69746785866511],
 'Salamanca': [40.4274452993663, -3.68749663492822],
 'Ciudad Lineal': [40.4218325688458, -3.65739134045678],
 'Centro': [40.4185779318042, -3.69693673318819],
 'Latina': [40.4135978394487, -3.72399602634882],
 'Carabanchel': [40.3851729900366, -3.7201628610672],
 'Hortaleza': [40.4624131597597, -3.62427096076886],
 'Puente De Vallecas': [40.3706754505008, -3.68039067757924],
 'Moratalaz': [40.4217163728431, -3.65725894396416],
 'Fuencarral-El Pardo': [40.5098738838586, -3.68517743671953],
 'Moncloa-Aravaca': [40.4290193658703, -3.72190583044638]}

In [115]:
# Creamos una columna con la latitud y longitud de cada distrito en el dataframe pivot_correlacion_transito_accidentes_bicicletas_2019

pivot_correlacion_transito_accidentes_bicicletas_2019['LATITUD'] = pivot_correlacion_transito_accidentes_bicicletas_2019.index.map(diccionario_distritos).str[0]
pivot_correlacion_transito_accidentes_bicicletas_2019['LONGITUD'] = pivot_correlacion_transito_accidentes_bicicletas_2019.index.map(diccionario_distritos).str[1]
pivot_correlacion_transito_accidentes_bicicletas_2019

ACCIDENTES  TRANSITO  PORCENTAJE_ACCIDENTES    LATITUD  \
DISTRITO                                                                      
Arganzuela                   46     69275              13.294798  40.393023   
Carabanchel                  22     34852               6.358382  40.385173   
Centro                       78     38990              22.543353  40.418578   
Chamberí                     39     62922              11.271676  40.428086   
Ciudad Lineal                15     40250               4.335260  40.421833   
Fuencarral-El Pardo          30     63075               8.670520  40.509874   
Hortaleza                    12     55620               3.468208  40.462413   
Latina                       13     33440               3.757225  40.413598   
Moncloa-Aravaca              24     14343               6.936416  40.429019   
Moratalaz                     9     49549               2.601156  40.421716   
Puente De Vallecas           19     40588               5.491329  40.370675   
Salamanca                    39     95438              11.271676  40.427445   

                     LONGITUD  
DISTRITO                       
Arganzuela          -3.697567  
Carabanchel         -3.720163  
Centro              -3.696937  
Chamberí            -3.697468  
Ciudad Lineal       -3.657391  
Fuencarral-El Pardo -3.685177  
Hortaleza           -3.624271  
Latina              -3.723996  
Moncloa-Aravaca     -3.721906  
Moratalaz           -3.657259  
Puente De Vallecas  -3.680391  
Salamanca           -3.687497

In [116]:
# Como tenemos la latitud y longitud de los distritos donde hay mas % de accidentes con bicicletas implicadas, vamos a hacer un mapa de calor para ver si hay alguna correlacion

import folium
from folium.plugins import HeatMap

mapa = folium.Map(location=[40.416775, -3.703790], zoom_start=12)

for i in range(len(pivot_correlacion_transito_accidentes_bicicletas_2019)):
    folium.CircleMarker(location=[pivot_correlacion_transito_accidentes_bicicletas_2019['LATITUD'][i], pivot_correlacion_transito_accidentes_bicicletas_2019['LONGITUD'][i]], radius=10, color='red', fill=True, fill_color='red', fill_opacity=0.5).add_to(mapa)

mapa

CONCLUSIONES EJERCICIO 6:

- Tenemos la zona con mas % de accidentes que en este caso es el centro y con un transito de bicicletas de 38.990 bicicletas.
- La zona con mas transito de bicicletas es el distrito de Arganzuela con 69.275 bicicletas pero con un % de accidentes con bicicletas implicadas de 13.29%.
- Moratalaz es el distrito con menos % de accidentes con bicicletas implicadas con un transito de bicicletas de 49.549 bicicletas.

**Ejercicio 7:** Por último, tambien disponemos de los accidentes en 2020 (`2020_Accidentalidad.xlsx`). Comparar la serie mensual del nº de accidentes en 2019 vs 2020 y ver cuanto se han reducidos los accidentes cada mes debido al efecto del confinamiento.

In [105]:
# Evolucion de los accidentes en 2019

px.line(df_2019.groupby('MES').count()['Nº  EXPEDIENTE'].reset_index(), x='MES', y='Nº  EXPEDIENTE')

In [106]:
# Evolucion de los accidentes en 2020

df_2020 = pd.read_excel('2020_Accidentalidad.xlsx')
df_2020.head()

Nº  EXPEDIENTE       FECHA      HORA  \
0    2020S000057  01/01/2020  23:15:00   
1    2020S000038  01/01/2020  22:35:00   
2    2020S000060  01/01/2020  20:15:00   
3    2020S000060  01/01/2020  20:15:00   
4    2020S000033  01/01/2020  19:45:00   

                                               CALLE NÚMERO  \
0  AVDA. CIUDAD DE BARCELONA / CALL. DOCTOR ESQUERDO      -   
1  CALL. VALLE DE TORANZO / CALL. SIERRA DE PAJAREJO      -   
2                       GTA. MARIANO SALVADOR MAELLA      1   
3                       GTA. MARIANO SALVADOR MAELLA      1   
4                                       CALL. OLIVAR     40   

              DISTRITO                TIPO ACCIDENTE ESTADO METEREOLÓGICO  \
0               RETIRO  Choque contra obstáculo fijo            Despejado   
1      MONCLOA-ARAVACA                         Caída            Despejado   
2  FUENCARRAL-EL PARDO                         Caída            Despejado   
3  FUENCARRAL-EL PARDO                         Caída            Despejado   
4               CENTRO  Choque contra obstáculo fijo            Despejado   

             TIPO VEHÍCULO TIPO PERSONA    RANGO DE EDAD    SEXO  LESIVIDAD*  \
0                  Turismo    Conductor  DE 25 A 29 AÑOS  Hombre         NaN   
1               Ciclomotor    Conductor  DE 21 A 24 AÑOS   Mujer         6.0   
2                  Turismo    Conductor  DE 45 A 49 AÑOS  Hombre        14.0   
3  Motocicleta hasta 125cc    Conductor  DE 25 A 29 AÑOS  Hombre         7.0   
4                  Turismo    Conductor      DESCONOCIDA     NaN         NaN   

   Unnamed: 13 Unnamed: 14  
0          NaN         NaN  
1          NaN         NaN  
2          NaN         NaN  
3          NaN         NaN  
4          NaN         NaN

In [107]:
# convertimos la columna FECHA a datetime

df_2020['FECHA'] = pd.to_datetime(df_2020['FECHA'])

/Users/adrianinfantesromero/opt/anaconda3/lib/python3.9/site-packages/pandas/core/tools/datetimes.py:1047: UserWarning:

Parsing '13/01/2020' in DD/MM/YYYY format. Provide format or specify infer_datetime_format=True for consistent parsing.

/Users/adrianinfantesromero/opt/anaconda3/lib/python3.9/site-packages/pandas/core/tools/datetimes.py:1047: UserWarning:

Parsing '14/01/2020' in DD/MM/YYYY format. Provide format or specify infer_datetime_format=True for consistent parsing.

/Users/adrianinfantesromero/opt/anaconda3/lib/python3.9/site-packages/pandas/core/tools/datetimes.py:1047: UserWarning:

Parsing '15/01/2020' in DD/MM/YYYY format. Provide format or specify infer_datetime_format=True for consistent parsing.

/Users/adrianinfantesromero/opt/anaconda3/lib/python3.9/site-packages/pandas/core/tools/datetimes.py:1047: UserWarning:

Parsing '16/01/2020' in DD/MM/YYYY format. Provide format or specify infer_datetime_format=True for consistent parsing.

/Users/adrianinfantesromero/opt/

In [108]:
# creamos una columna MES

df_2020['MES'] = df_2020['FECHA'].dt.month

In [109]:
# Veamos ahora la evolucion de 2020

px.line(df_2020.groupby('MES').count()['Nº  EXPEDIENTE'].reset_index(), x='MES', y='Nº  EXPEDIENTE', title='Evolución de los accidentes en 2020')

**Ejercicio casa:** Validar las siguientes conclusiones ([datos](https://datos.madrid.es/portal/site/egob/menuitem.c05c1f754a33a9fbe4b2e4b284f1a5a0/?vgnextoid=20f4a87ebb65b510VgnVCM1000001d4a900aRCRD&vgnextchannel=374512b9ace9f310VgnVCM100000171f5a0aRCRD&vgnextfmt=default/)):

[artículo 1](https://www.elconfidencial.com/arcadia/2022-06-09/mapa-accidentes-madrid_3438845/)

[artículo 2](https://www.abc.es/espana/madrid/madrid-ciudad-bicis-accidentes-caos-carriles-ciclistas-20220808202236-nt.html?ref=https%3A%2F%2Fwww.google.com%2F)

## Concluiones generales:

- Los hombres están implicados en más accidentes que las mujeres.
- Los hombres están implicados en más accidentes que las mujeres en todos los rangos de edad menos en el rango de edad DE 10 A 14 AÑOS.
- En el rango de edad DE 10 A 14 AÑOS las mujeres están implicadas en más accidentes que los hombres.
- En el rango de edad DE 10 A 14 AÑOS las mujeres que estaban implicadas en accidentes eran: conductores, peatones y pasajeros.
- En el rango de edad DE 10 A 14 AÑOS las mujeres que estaban implicadas en accidentes como conductores eran un total de 8, 6 de ellas conducían bicicletas y 1 conducía algo a motor y por ultimo 1 de ellas sin especificar el tipo de vehículo. Sospechoso.... no?
- El día del año con más accidentes fue el dia 124 del año 2019. Es decir el 23 de abril.
- El número de accidentes no es constante a lo largo de los meses del año 2019. Vemos que en los meses 5 y 11 son los picos de accidentes mientras que los meses 2 y 8 son los
meses con menos accidentes.
- El distrito con más accidentes es SALAMANCA.
- Las 7 calles con más accidentes son: PASEO. CASTELLANA, CALL. ALCALA, CALL. BRAVO MURILLO, AVDA. ALBUFERA, CALL. SERRANO, CALL. PRINCIPE DE VERGARA, PASEO. SANTA MARIA DE LA CABEZA.
- La calle con más accidentes es PASEO. CASTELLANA con 6.300 metros de longitud y 309 accidentes.
- La calle con menos accidentes es PASEO. SANTA MARIA DE LA CABEZA con 3.300 metros de longitud y 77 accidente.
- Tenemos la zona con mas % de accidentes que en este caso es el centro y con un transito de bicicletas de 38.990 bicicletas.
- La zona con mas transito de bicicletas es el distrito de Arganzuela con 69.275 bicicletas pero con un % de accidentes con bicicletas implicadas de 13.29%.
- Moratalaz es el distrito con menos % de accidentes con bicicletas implicadas con un transito de bicicletas de 49.549 bicicletas.